src/libraries/System.Text.Json/tests/JsonNode/JsonValueTests.cs

```diff
@@ -0,0 +1,291 @@
+
+        [Fact]
+        public static void Polymorphic()
+        {
+            JsonValue value;
+
+            Polymorphic_Base baseClass = new Polymorphic_Derived();
+            value = JsonValue.Create(baseClass);
+            Assert.Same(baseClass, value.GetValue<Polymorphic_Derived>());
+
+            Polymorphic_Derived derivedClass = new Polymorphic_Derived();
+            value = JsonValue.Create(derivedClass);
+            Assert.Same(derivedClass, value.GetValue<Polymorphic_Base>());
+
+            Assert.Same(derivedClass, value.GetValue<object>());
+        }
+
+        [Fact]
+        public static void QuotedNumbers_Deserialize()
+        {
+            var options = new JsonSerializerOptions();
+            options.NumberHandling = JsonNumberHandling.AllowReadingFromString |
+                JsonNumberHandling.AllowNamedFloatingPointLiterals;
+
+            JsonNode node = JsonSerializer.Deserialize<JsonNode>("\"42\"", options);
+            Assert.IsAssignableFrom<JsonValue>(node);
+            Assert.Throws<InvalidOperationException>(() => node.GetValue<int>());
+
+            // A second pass is needed to obtain the quoted number.
+            Assert.Equal(42, JsonSerializer.Deserialize<int>(node.ToJsonString(), options));
+
+            node = JsonSerializer.Deserialize<JsonNode>("\"NaN\"", options);
+            Assert.IsAssignableFrom<JsonValue>(node);
+            Assert.Equal(double.NaN, JsonSerializer.Deserialize<double>(node.ToJsonString(), options));
+            Assert.Equal(float.NaN, JsonSerializer.Deserialize<float>(node.ToJsonString(), options));
+        }
+
+        [Fact]
+        public static void QuotedNumbers_Serialize()
+        {
+            var options = new JsonSerializerOptions();
+            options.NumberHandling = JsonNumberHandling.WriteAsString;
+
+            JsonValue obj = JsonValue.Create(42);
+            string json = obj.ToJsonString(options);
+            Assert.Equal("\"42\"", json);
+
+            obj = JsonValue.Create(double.NaN);
+            json = obj.ToJsonString(options);
+            Assert.Equal("\"NaN\"", json);
+        }
+
+        [Fact]
+        public static void FromElement()
+        {
+            using (JsonDocument document = JsonDocument.Parse("42"))
+            {
+                JsonValue jValue = JsonValue.Create(document.RootElement);
+                Assert.Equal(42, jValue.GetValue<int>());
+            }
+
+            using (JsonDocument document = JsonDocument.Parse("null"))
+            {
+                JsonValue? jValue = JsonValue.Create(document.RootElement);
+                Assert.Null(jValue);
+            }
+        }
+
+        [Fact]
+        public static void TryGetValue_FromString()
+        {
+            JsonValue jValue = JsonNode.Parse("\"MyString\"").AsValue();
+
+            Assert.True(jValue.TryGetValue(out string _));
+            Assert.True(jValue.TryGetValue(out char _));
```

Comment:
Should `char` be passing in this case (asking per `CharConverter` semantics)? The string is more than one char long. Here and in the tests below...

Reply:
Yes, thanks

Reply:
It should be changed to match the serializer semantics. Thanks